components/localeswitcherselect.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+
+import clsx from "clsx";
+import { ChangeEvent, ReactNode, useTransition } from "react";
+import { usePathname, useRouter } from "../lib/i18n/navigation";
+
+type Props = {
+  children: ReactNode;
+  defaultValue: string;
+  label: string;
+};
+
+export default function LocaleSwitcherSelect({
+  children,
+  defaultValue,
+  label,
+}: Props) {
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const pathname = usePathname();
+
+  function onSelectChange(event: ChangeEvent<HTMLSelectElement>) {
+    const nextLocale = event.target.value;
+    startTransition(() => {
+      router.replace(pathname, { locale: nextLocale });
+    });
+  }
+
+  return (
+    <label
+      className={clsx(
+        "relative text-gray-400",
+        isPending && "transition-opacity [&:disabled]:opacity-30"
+      )}
+    >
+      <p className="sr-only">{label}</p>
+      <select
+        className="inline-flex appearance-none bg-transparent py-3 pl-2 pr-6"
+        defaultValue={defaultValue}
+        disabled={isPending}
+        onChange={onSelectChange}
+      >
+        {children}
+      </select>
```

Comment:
I'll write it here even if it is not about the component itself:
I don't think we should translate language names. I mean, if I'm using Woel in `English-GB`, I guess it would make more sense to find `Italiano-IT` in the language select (currently, if I accidentally set a foreign language, it might be hard to revert the change)

Reply:
currently it looks like this... in this case the current lang is English
<img width="342" alt="image" src="https://github.com/ferrariofilippo/woel/assets/79911341/41b6436a-4adb-49c2-8a91-a77ea65b1482">